expression/constant.go

```diff
@@ -112,8 +112,9 @@ type Constant struct {
 	DeferredExpr Expression
 	// ParamMarker holds param index inside sessionVars.PreparedParams.
 	// It's only used to reference a user variable provided in the `EXECUTE` statement or `COM_EXECUTE` binary protocol.
-	ParamMarker *ParamMarker
-	hashcode    []byte
+	ParamMarker               *ParamMarker
+	hashcode                  []byte
+	ignoreParamMarkerHashCode []byte // use ConstantFlag+EncodedValue even if ParamMarker is not nil
```

Comment:
Why do we need a new one, can we reuse the `hashcode`?

Reply:
Revert the change now. It is somewhat like the "canonicalhashcode" in scalar function. Different goals lead to different hash code. However, also need to consider add it in scalar function level to avoid the cache mechanism, so revert it currently.